File: 0349-Intersection-of-Two-Arrays/0349.py
class Solution:
    def intersection(self, nums1, nums2):
        """
        :type nums1: List[int]
        :type nums2: List[int]
        :rtype: List[int]
        """
        result = set([i for i in nums1 if i in nums2])
        return list(result)

if __name__ == "__main__":
    nums1 = [1, 2, 2, 1]
    nums2 = [2, 2]
    print(Solution().intersection(nums1, nums2))

File: src/0349-Intersection-of-Two-Arrays/0349.cpp
#include <unordered_set>
#include <iostream>
#include <vector>
using namespace std;

static int x = []() {std::ios::sync_with_stdio(false); cin.tie(0); return 0; }();
class Solution 
{
public:
    vector<int> intersection(vector<int>& nums1, vector<int>& nums2) 
    {
        unordered_set<int> record(nums1.begin(), nums1.end());
        unordered_set<int> result;
        for (auto& i : nums2)
        {
            if (record.count(i) != 0) result.insert(i);
        }
        return vector<int>(result.begin(), result.end());
    }
};
int main()
{
    vector<int> nums1 = { 1, 2, 2, 1 };
    vector<int> nums2 = { 2, 2 };
    vector<int> result = Solution().intersection(nums1, nums2);
    for (auto& i : result)
    {
        cout << i << " ";
    }
    cout << endl;
    return 0;
}